### PR TITLE
build internal compiler with opensource tao_build.py

### PR DESCRIPTION
**Pull request by Orion34-lanbo**
Since internal `tao_build.py` share a lot of duplicate codes as the one in current repo, we chose to merge the build the logic into current `tao_build.py`.

### CODE REVIEW

File: scripts/python/tao_build.py [hunk 1]
```diff
@@ -482,6 +527,8 @@ def bazel_test(target, flag=""):
                 flag = '--config=disc_x86 '
             if args.enable_mkldnn:
                 flag += ' --config=disc_mkldnn'
+            if args.platform_alibaba:
```
Comment (qiuxiafei) [inline scripts/python/tao_build.py:530]: flags here seems similar to those starts from L368, can they get merged?

File: scripts/python/tao_build.py [hunk 2]
```diff
@@ -692,6 +760,54 @@ def get_libstdcxx(gcc_ver):
     return None, None
 
 
+@time_stage()
+def make_package(root, args):
+    libstdcxx_path, libstdcxx_name = get_libstdcxx(args.compiler_gcc)
```
Comment (qiuxiafei) [inline scripts/python/tao_build.py:765]: why're we going to pack libstdc+`+.so` ?

File: tao/BUILD [hunk 3]
```diff
@@ -72,6 +73,19 @@ cc_proto_library(
     ],
 )
 
+filegroup(
+    name = "exclude_srcs",
```
Comment (qiuxiafei) [inline tao/BUILD:77]: not used?

File: tao/BUILD [hunk 4]
```diff
@@ -80,6 +94,7 @@ cc_library(
         ],
         exclude = [
             "tao_bridge/**/*test.cc",
+            "tao_bridge/gpu/bace_gemm_thunk.cc",
```
Comment (qiuxiafei) [inline tao/BUILD:97]: Just remove bace code internal code base?

File: third_party/bazel/blade_gemm/blade_gemm.BUILD.tpl [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+filegroup(
+    name = "all_source_files",
+    srcs = glob(["**"]),
+    visibility = ["//visibility:private"],
+)
+
+cmake(
+    name = "blade_gemm",
+    cache_entries = {
+        "CMAKE_BUILD_TYPE": "Release",
```
Comment (qiuxiafei) [inline third_party/bazel/blade_gemm/blade_gemm.BUILD.tpl:16]: nit: if binary size is not a concern, `RelWithDebInfo` is preffered.

File: scripts/python/common_setup.py [hunk 6]
```diff
@@ -216,6 +216,10 @@ def remote_cache_token():
     if os.path.exists(fn):
         with open(fn) as f:
             return str(f.read()).strip()
+    else:
```
[the remark this answers is not in the document]
Comment (Orion34-lanbo) [reply scripts/python/common_setup.py:219]: Internal env does not have this file.. so we check the env

File: third_party/bazel/blade_gemm/blade_gemm.BUILD.tpl [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        "CMAKE_CXX_COMPILER": cxx_bin_path(),
+        "CUTLASS_NVCC_ARCHS": "%{CUTLASS_NVCC_ARCHS}",
+        "CUTLASS_LIBRARY_KERNELS": "%{CUTLASS_LIBRARY_KERNELS}",
+        "CUTLASS_ENABLE_TESTS": "OFF",
+        "CUTLASS_UNITY_BUILD_ENABLED": "ON",
+    },
+    env = {
+        "CC": cc_bin_path(),
+        "CXX": cxx_bin_path(),
+        "CUDACXX": "%{CUTLASS_CUDACXX}",
```
Comment (qiuxiafei) [inline third_party/bazel/blade_gemm/blade_gemm.BUILD.tpl:28]: `BLADE_GEMM_NVCC`?

File: third_party/bazel/blade_gemm/blade_gemm_configure.bzl [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+            {
+                "%{BLADE_GEMM_REPO_PATH}": get_host_environ(repository_ctx, _BLADE_GEMM_REPO_PATH, "../../platform_alibaba/blade_gemm/"),
+            },
+        )
+
+        repository_ctx.template("blade_gemm.BUILD", Label("//bazel/blade_gemm:blade_gemm.BUILD.tpl"),
+            {
+                "%{CUTLASS_NVCC_ARCHS}": get_host_environ(repository_ctx, _CUTLASS_NVCC_ARCHS, "80"),
+                "%{CUTLASS_LIBRARY_KERNELS}": get_host_environ(repository_ctx, _CUTLASS_LIBRARY_KERNELS, "s1688tf32gemm,f16_s1688gemm_f16,f16_s16816gemm_f16,s16816tf32gemm"),
+                "%{CUTLASS_CUDACXX}": get_host_environ(repository_ctx, _CUTLASS_CUDACXX, "/usr/local/cuda-11.6/bin/nvcc"),
```
Comment (qiuxiafei) [inline third_party/bazel/blade_gemm/blade_gemm_configure.bzl:24]: no default?

File: third_party/bazel/blade_gemm/blade_gemm.BUILD.tpl (hunk 5 above)
Comment (Orion34-lanbo) [reply third_party/bazel/blade_gemm/blade_gemm.BUILD.tpl:16]: libblade_gemm.a has a size of 66MB, I think it is quite big already.

File: tao/BUILD (hunk 4 above)
Comment (Orion34-lanbo) [reply tao/BUILD:97]: I add a macro to not use `bace` related code now.

File: scripts/python/tao_build.py (hunk 2 above)
Comment (Orion34-lanbo) [reply scripts/python/tao_build.py:765]: These are just copied from internal repo.. maybe useful for non-devtoolset build.

File: scripts/python/common_setup.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (qiuxiafei) [reply scripts/python/common_setup.py:219]: at least unibuild has followed the former...

File: tao/BUILD (hunk 4 above)
Comment (qiuxiafei) [reply tao/BUILD:97]: Great

File: scripts/python/tao_build.py [hunk 9]
```diff
@@ -318,9 +318,16 @@ def _write(line, cmd="build"):
         is_cuda = not (args.cpu_only or args.dcu)
         if is_cuda:
             cuda_ver, _ = deduce_cuda_info()
-            if '11\.' in cuda_ver:
+            logger.info(f"Builing with cuda-{cuda_ver}")
+            if '11.' in cuda_ver:
```
Comment (qiuxiafei) [inline scripts/python/tao_build.py:322]: nit: `cuda_ver.startswith('11.')` seems better.

File: scripts/python/tao_build.py (hunk 2 above)
Comment (qiuxiafei) [reply scripts/python/tao_build.py:772]: Fine, let it be...

**Pull request merged**